infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/pvc/CommonPVCStrategy.java

```diff
@@ -258,4 +274,70 @@ private PersistentVolumeClaim replacePVCsWithCommon(
         .filter(subpath -> !isNullOrEmpty(subpath))
         .collect(Collectors.toSet());
   }
+
+  private void deleteCommonPVC(Workspace workspace) throws InfrastructureException {
+    factory.get(workspace).persistentVolumeClaims().delete(configuredPVCName);
+  }
+
+  /**
+   * @return true, if the common PVC is expected to be deleted, false otherwise. Depending on the
+   *     configuration it could be either the common PVC of a particular user, or the common PVC
+   *     that is shared across all the users (the last setup is considered to be uncommon and
+   *     not-recommended)
+   */
+  private boolean noWorkspacesLeft() {
+    return defaultNamespaceWithoutPlaceholderIsDefined()
+        ? totalNumberOfWorkspacesIsZero()
+        : userHasNoWorkspaces();
+  }
+
+  /**
+   * The method is expected to be used in order to identify if the common PVC is used across all the
+   * users. The common PVC for all the users will be used if
+   * 'che.infra.kubernetes.namespace.default' points to a particular namespace e.g. 'che',
+   * 'workspaces' etc.
+   *
+   * @return true, if 'che.infra.kubernetes.namespace.default' is defined and does NOT contain the
+   *     placeholder e.g.: che-workspace-<username>), false otherwise
+   */
+  private boolean defaultNamespaceWithoutPlaceholderIsDefined() {
+    return defaultNamespaceName != null
+        && !defaultNamespaceName.contains("<")
+        && !defaultNamespaceName.contains(">");
+  }
+
+  /** @return true, if there are no workspaces left across all the users, false otherwise */
+  private boolean totalNumberOfWorkspacesIsZero() {
+    try {
+      return workspaceManager.getWorkspacesTotalCount() == 0;
+    } catch (ServerException e) {
+      log.error("Unable to get the total number of workspaces. Cause: {}", e.getMessage(), e);
```

Comment:
why this error has been suppressed?

Reply:
technically, this error should never happen. If it happens we just log error and return false (do not delete PVC)

Reply:
Should I throw a RuntimeException here to be consistent with https://github.com/eclipse-che/che-server/pull/16#discussion_r641895327?  Or maybe keep `log.error()` for both cases?